cirq-core/cirq/circuits/circuit.py

```diff
@@ -1968,6 +1989,8 @@ def _prev_moment_available(self, op: 'cirq.Operation', end_moment_index: int) ->
             ):
                 return last_available
             if self._can_add_op_at(k, op):
+                # Note: Remove the if condition after `self._device` is gone and move the method to
```

Comment:
Not sure I understand this comment - `cirq.AbstractDevice` is for post-processing of the circuit, right?

Reply:
`_can_add_op_at` will be moot here once `self._device` is being deprecated -- i.e. circuits will stop using devices for validation, which is being done as part of the devices 2.0 migration (see https://github.com/quantumlib/Cirq/issues/4744)